lib/widgets/screens/inbox/channel_messages/channel_messages_screen.dart

```diff
@@ -123,65 +121,70 @@ class _ChannelChatState extends State<ChannelChat>
     _chatModel = Provider.of<ChatModel>(context, listen: false);
     _chatModel.markMessagesAsRead();
     _chatModel.createChannelSubscription();
-    _messageCount = _chatModel.channelMessages.length;
+    _chatModel.onNewMessage = _processNewMessages;
+    listScrollController.addListener(_fetchOlderMessages);
   }
 
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
-    _chatModel.refreshChannelMessages();
+    _chatModel.addOldChannelMessages();
   }
 
   void _processNewMessages() {
-    if (_messageCount < _chatModel.channelMessages.length) {
-      _chatModel.markMessagesAsRead();
-      if (_chatModel.channelMessages.last.createdBy ==
-          _chatModel.authenticatedUser.id) {
-        _scrollDown();
-      } else if (_unreadMessages != true) {
-        _unreadMessages = true;
-      }
+    _chatModel.markMessagesAsRead();
+    if (_chatModel.channelMessages.last.createdBy ==
+        _chatModel.authenticatedUser.id) {
+      _scrollDown();
+    }
+  }
+
+  void _fetchOlderMessages() {
+    final extentAfter = listScrollController.position.extentAfter;
+    if (extentAfter < 300 &&
```

Comment:
When does the scroll listener get triggered to load more messages? When there are less than 300 pixels between the top of the content (which is hidden) and the visible screen, we load 25 more messages?

Dug through the Flutter docs a bit (ScrollListener, ScrollPosition, the [extentAfter property](https://api.flutter.dev/flutter/widgets/ScrollMetrics/extentAfter.html)) to try to understand this, but I'm still not 100% sure I know what's happening here.

Reply:
Yes that is pretty much what is happening. The scrollable content goes beyond the visible top of the screen, and this function triggers when there is less than 300 pixels to reach the top of the scrollable area. Once more messages load, the top of the scrollable area gets pushed up further so the user has to keep scrolling to trigger this function again.

The number 300 is sort of a magic number I arrived to when testing. Much lower than this and scrolling won't be smooth, much higher than this and the loading gets triggered too easily and sometimes more than once since the number of messages fetched doesn't push the scrollable area up enough.